lib/debugger.coffee

```diff
@@ -23,19 +23,28 @@ class ProcessManager extends EventEmitter
     result[key(e)] = value(e) for e in env.split(";")
     return result
 
-  start: (file) ->
+  startActiveFile: () ->
+    @start true
+
+  start: (withActiveFile) ->
+    startActive = withActiveFile
     @cleanup()
       .then =>
+        packageJSON = require @atom.project.resolvePath('package.json')
         nodePath = @atom.config.get('node-debugger.nodePath')
         nodeArgs = @atom.config.get('node-debugger.nodeArgs')
         appArgs = @atom.config.get('node-debugger.appArgs')
         port = @atom.config.get('node-debugger.debugPort')
         env = @parseEnv @atom.config.get('node-debugger.env')
+        scriptMain = @atom.project.resolvePath(@atom.config.get('node-debugger.scriptMain'))
+
+        dbgFile = scriptMain || packageJSON && @atom.project.resolvePath(packageJSON.main)
 
-        editor = @atom.workspace.getActiveTextEditor()
-        appPath = editor.getPath()
+        if startActive == true || !dbgFile
```

Comment:
if `dbgFile` still isnt' a thing (no script main, or package.json main), try to debug the active file